File: source/downloads/code/LeetCode/Java/Remove-Duplicates-from-Sorted-List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if (head == null)
            return head;
        ListNode prev = head;
        ListNode nextNode = prev.next;
        while (nextNode != null){
            while (nextNode != null && nextNode.val == prev.val)
                nextNode = nextNode.next;
            prev.next = nextNode;
            prev = prev.next;
        }
        return head;  
    }
}

File: downloads/code/LeetCode/Python/Remove-Duplicates-from-Sorted-List.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if head == None:
            return head
	newHead = ListNode(0)
        newHead.next = head
        ret = newHead

        newHead = head
        head = head.next

        while head != None:
            if head.val != newHead.val:
                newHead.next = head
                newHead = head
            head = head.next
        newHead.next = None
        return ret

